src/Components/Reusable/Send/03-SummarySend/Components/TransactionFeeCard.tsx

```diff
@@ -0,0 +1,252 @@
+                dispatch(setIsAppLoading(false))
+                onFinish(true)
+            } catch (e) {
+                error(ERROR_EVENTS.SEND, e)
+                onFinish(false)
+            }
+        },
+        [dispatch, onFinish],
+    )
+
+    const onTransactionFailure = useCallback(() => {
+        onFinish(false)
+    }, [onFinish])
+
+    const clauses = useMemo(
+        () => TransactionUtils.prepareFungibleClause(finalAmount, token, address),
+        [finalAmount, token, address],
+    )
+
+    const {
+        selectedDelegationOption,
+        onSubmit,
+        isPasswordPromptOpen,
+        handleClosePasswordModal,
+        onPasswordSuccess,
+        setSelectedFeeOption,
+        resetDelegation,
+        setSelectedDelegationAccount,
+        setSelectedDelegationUrl,
+        isEnoughGas,
+        isDelegated,
+        selectedDelegationAccount,
+        selectedDelegationUrl,
+        isDisabledButtonState,
+        gasOptions,
+        gasUpdatedAt,
+        selectedFeeOption,
+        isGalactica,
+        isBaseFeeRampingUp,
+        speedChangeEnabled,
+        availableTokens,
+        selectedDelegationToken,
+        setSelectedDelegationToken,
+        fallbackToVTHO,
+        hasEnoughBalanceOnAny,
+        isFirstTimeLoadingFees,
+        hasEnoughBalanceOnToken,
+    } = useTransactionScreen({
+        clauses,
+        onTransactionSuccess,
+        onTransactionFailure,
+        autoVTHOFallback: false,
+    })
+
+    const [hasReportedGasAdjustment, setHasReportedGasAdjustment] = useState(false)
+
+    useEffect(() => {
+        if (!onBindTransactionControls) return
+
+        onBindTransactionControls({
+            onSubmit,
+            isDisabledButtonState,
+        })
+    }, [onBindTransactionControls, onSubmit, isDisabledButtonState])
+
+    /**
+     * If user is sending a token and gas is not enough, we will adjust the amount to send or switch fee token.
+     */
+    useEffect(() => {
+        if (isDelegated && selectedDelegationToken === VTHO.symbol) {
+            return
+        }
+        if (isEnoughGas) {
+            return
+        }
+        if (selectedDelegationToken.toLowerCase() !== token.symbol.toLowerCase()) {
+            fallbackToVTHO()
+            if (!hasReportedGasAdjustment && typeof onGasAdjusted === "function") {
+                onGasAdjusted()
+                setHasReportedGasAdjustment(true)
+            }
+            return
+        }
+
+        const gasFees = gasOptions[selectedFeeOption].maxFee
+        const balance = BigNutils(token.balance.balance)
+        if (BigNutils(amount).multiply(BigNutils(10).toBN.pow(18)).plus(gasFees.toBN).isBiggerThan(balance.toBN)) {
+            const newBalance = balance.minus(gasFees.toBN)
+            if (newBalance.isLessThanOrEqual("0")) {
+                fallbackToVTHO()
+                setFinalAmount(amount)
+            } else setFinalAmount(newBalance.toHuman(token.decimals).decimals(4).toString)
```

Comment:
Same here, I wouldn't apply this logic.
I'd try to keep the token decimals as much as possible and show it on the UI properly

Reply:
Agree, truncating there was the wrong layer/place to do formatting. Refactored